### PR TITLE
fix: React dom failure when passing unknown props to child component

### PR DESCRIPTION
**Pull request by memoyil**
Screenshot 2021-04-02 at 13 35 56

### CODE REVIEW

File: packages/pancake-uikit/src/components/ButtonMenu/ButtonMenuItem.tsx [hunk 1]
```diff
@@ -6,12 +6,12 @@ import { ButtonMenuItemProps } from "./types";
 
 interface InactiveButtonProps extends BaseButtonProps {
   forwardedAs: BaseButtonProps["as"];
-  colorKey: "primary" | "textSubtle";
+  colorkey: "primary" | "textSubtle";
 }
 
 const InactiveButton: PolymorphicComponent<InactiveButtonProps, "button"> = styled(Button)<InactiveButtonProps>`
   background-color: transparent;
-  color: ${({ theme, colorKey }) => theme.colors[colorKey]};
```
Comment (RabbitDoge) [inline packages/pancake-uikit/src/components/ButtonMenu/ButtonMenuItem.tsx:14]: The initial solution here is wrong imo. If you want to clean this, a better solution is to remove the `colorkey` attribute and compute the color based on the variant attribute

Comment (hachiojidev) [reply packages/pancake-uikit/src/components/ButtonMenu/ButtonMenuItem.tsx:14]: @memoyil 

@RabbitDoge is right in it is not an ideal solution. Do you mind fixing it? No worries if you can't, I will get to it.

Comment (memoyil) [reply packages/pancake-uikit/src/components/ButtonMenu/ButtonMenuItem.tsx:14]: @hachiojidev I will try to fix it :) if not I will ask your help

Comment (memoyil) [reply packages/pancake-uikit/src/components/ButtonMenu/ButtonMenuItem.tsx:14]: @hachiojidev can you review it again?

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**